cmd/terramate/e2etests/safeguard_test.go

```diff
@@ -118,94 +121,97 @@ func TestSafeguardFailsOnRunIfRemoteMainIsOutdated(t *testing.T) {
 func TestSafeguardDisableGitCheckRemote(t *testing.T) {
 	t.Parallel()
 
-	s := sandbox.New(t)
-
-	stack := s.CreateStack("stack")
 	fileContents := "# whatever"
-	someFile := stack.CreateFile("main.tf", fileContents)
 
-	tmcli := newCLI(t, s.RootDir())
+	setup := func(t *testing.T) (tmcli, sandbox.FileEntry, sandbox.S) {
+		t.Helper()
+		s := sandbox.New(t)
 
-	git := s.Git()
+		stack := s.CreateStack("stack")
+		someFile := stack.CreateFile("main.tf", fileContents)
 
-	git.Add(".")
-	git.Commit("all")
+		git := s.Git()
+		git.CommitAll("all")
 
-	setupLocalMainBranchBehindOriginMain(git, func() {
-		stack.CreateFile("some-new-file", "testing")
-	})
+		setupLocalMainBranchBehindOriginMain(git, func() {
+			stack.CreateFile("some-new-file", "testing")
+		})
+
+		return newCLI(t, s.RootDir()), someFile, s
+	}
 
 	cat := test.LookPath(t, "cat")
 
-	t.Run("check remote is not reachable", func(t *testing.T) {
+	t.Run("make sure setup() makes origin/main unreachable", func(t *testing.T) {
+		tmcli, file, _ := setup(t)
 		assertRunResult(t, tmcli.run(
 			"run",
 			cat,
-			someFile.HostPath(),
+			file.HostPath(),
 		),
 			runExpected{
 				Status:      1,
 				StderrRegex: string(cli.ErrCurrentHeadIsOutOfSync),
 			})
 	})
 
-	t.Run("disable check using cmd args", func(t *testing.T) {
+	t.Run("disable check_remote safeguard using --disable-check-git-remote", func(t *testing.T) {
+		tmcli, file, _ := setup(t)
```

Comment:
much better to extract the setup and not share state between the tests, thanks :100: